stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/message/list/MessageListController.kt

```diff
@@ -1405,23 +1405,43 @@ public class MessageListController(
         ).enqueue { result ->
             if (result.isSuccess) {
                 val message = result.data()
+
                 message.attachments.removeAll { attachment ->
-                    if (attachmentToBeDeleted.assetUrl != null) {
-                        attachment.assetUrl == attachmentToBeDeleted.assetUrl
-                    } else {
-                        val isSame = attachment.imageUrl == attachmentToBeDeleted.imageUrl
-                        isSame
+                    val imageUrl = attachmentToBeDeleted.imageUrl
+                    val assetUrl = attachmentToBeDeleted.assetUrl
+
+                    when {
+                        assetUrl != null -> {
+                            attachment.assetUrl?.substringBefore("?") ==
```

Comment:
`substringBefore()` will return the original string if it cannot satisfy this condition.

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/substring-before.html